src/Features/Core/Portable/Formatting/FormattingDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Formatting
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    internal class FormattingDiagnosticAnalyzer
+        : AbstractCodeStyleDiagnosticAnalyzer
+    {
+        public static readonly string ReplaceTextKey = nameof(ReplaceTextKey);
+
+        public static readonly ImmutableDictionary<string, string> RemoveTextProperties =
+            ImmutableDictionary.Create<string, string>().Add(ReplaceTextKey, "");
+
+        public FormattingDiagnosticAnalyzer()
+            : base(
+                IDEDiagnosticIds.FormattingDiagnosticId,
+                new LocalizableResourceString(nameof(FeaturesResources.Formatting_analyzer_title), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+                new LocalizableResourceString(nameof(FeaturesResources.Formatting_analyzer_message), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SyntaxAnalysis;
+
+        public override bool OpenFileOnly(Workspace workspace)
+            => false;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxTreeAction(AnalyzeSyntaxTree);
+
+        private void AnalyzeSyntaxTree(SyntaxTreeAnalysisContext context)
+        {
+            if (!(context.Options is WorkspaceAnalyzerOptions workspaceAnalyzerOptions))
```

Comment:
I am curious how the analyzer works on command line builds as there is no workspace.

Reply:
Synced up with @sharwell offline, and the approach for preview1 is to package the workspaces assemblies with the analyzer NuGet package, so it works on command line. We should fix this soon to avoid requiring this workaround.

Reply:
I think he is going to create a console app with Workspace dependency. once we have it, I think we make a lot of thing possible in the command line scenario.

glad that we finally have a place for such tool.

Reply:
> I am curious how the analyzer works on command line builds as there is no workspace.

The command line handling is now implemented in the CodeStyle layer packages.